src/ehttpc.erl

```diff
@@ -556,22 +561,6 @@ mk_reqopts(undefined) ->
 mk_reqopts(TunnelRef) ->
     #{tunnel => TunnelRef}.
 
-cancel_stream(fin, _Client, _StreamRef) ->
-    %% nothing to cancel anyway
-    %% otherwise gun will reply with a gun_error messsage
-    %% which is then discarded anyway
-    ok;
-cancel_stream(nofin, Client, StreamRef) ->
-    %% this is just an async message sent to gun
-    %% the gun stream process does not really cancel
-    %% anything, but just mark the receiving process (i.e. self())
-    %% as inactive, however, there could be messages already
-    %% delivered to self()'s mailbox
-    %% or the stream process might send more messages
-    %% before receiving the cancel message.
-    _ = gun:cancel(Client, StreamRef),
```

Comment:
Q: since we now send just `{error, expired}` to the callback, without the late response, shouldn't we still cancel the stream?